### ISSUE TITLE
How should Node require avoid the `d3.event` global?

### ISSUE THREAD

**Issue by mbostock** [d3/d3-selection#13]
Event handling requires setting `d3.event`, which therefore depends on a `d3` global. That global is defined in standalone.js. However, it would be nice if the `d3` object did not have to be global, as when the library is loaded via RequireJS.

Related, the standalone.js should do something like this:

```js
if (typeof define === "function" && define.amd) define(d3);
else if (typeof module === "object" && module.exports) module.exports = d3;
global.d3 = d3;
```

… except of course that doesn’t work with Browserify because it defines module.exports.

**Comment by gmaclennan** [comment]
What about using `browserify --standalone d3`? Then you would write `standalone.js` as:

```js
var event = require("./lib/d3/event"),
    namespace = require("./lib/d3/namespace"),
    selection = require("./lib/d3/selection"),
    d3;

d3.mouse = event.mouse;
d3.touch = event.touch;
d3.touches = event.touches;
d3.namespace = namespace.prefix;
d3.selection = selection;
d3.select = selection.select;
d3.selectAll = selection.selectAll;

// Deprecated aliases for backwards-compatibility with 3.x:
d3.ns = namespace;
selection.prototype.on = selection.prototype.event;
selection.prototype.insert = selection.prototype.append;
selection.prototype.classed = selection.prototype.class;

module.exports = d3;
```

`browserify --standalone d3` output would include a UMD and would be compatible with RequireJS etc.

**Comment by mbostock** [comment]
Yes. See #26.

**Comment by mbostock** [comment]
Mostly fixed by #30, but event listeners still depend on the `d3` global, and I’m not sure how to fix that.

**Comment by mbostock** [comment]
Actually, since `d3.event` is only needed within the context of an event listener, it seems fine as-is.
